api/classes/class-lightweight-api.php

```diff
@@ -224,23 +285,31 @@ public function get_client_data( $client_id, $do_not_rebuild = false ) {
 			$wpdb->prepare( "SELECT * FROM $events_table_name WHERE client_id = %s AND type = 'post_read'", $client_id ) // phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared
 		);
 
+		// If client hasn't read any posts yet, don't write any data.
+		if ( 0 === count( $client_post_read_events ) ) {
+			return $this->client_data_blueprint;
+		}
+
 		$this->save_client_data(
 			$client_id,
-			[
-				'posts_read' => array_map(
-					function ( $item ) {
-						return [
-							'post_id'      => $item->post_id,
-							'category_ids' => $item->category_ids,
-							'created_at'   => $item->created_at,
-						];
-					},
-					$client_post_read_events
-				),
-			]
+			array_merge(
```

Comment:
The merge is not necessary here, the blueprint will be merged in during the save here:

https://github.com/Automattic/newspack-popups/blob/a18ae51fc405f3f6a1b42207510330f6afde6a51/api/classes/class-lightweight-api.php#L345

& 

https://github.com/Automattic/newspack-popups/blob/a18ae51fc405f3f6a1b42207510330f6afde6a51/api/classes/class-lightweight-api.php#L290

I've added a test to demonstrate an issue with merging it twice:

https://github.com/Automattic/newspack-popups/blob/a18ae51fc405f3f6a1b42207510330f6afde6a51/tests/test-api.php#L595-L598
 
After the merging of blueprint is removed, the test will pass.

Reply:
You're right, thanks for the new test and for explaining why the merge isn't needed! Updated in dfc3b19.